### ISSUE TITLE
Something wrong with continuations, parameters, or dynamic-wind

### ISSUE THREAD

**Issue by sabauma** [samth/pycket#196]
The underlying cause of this problem is somewhat unclear, but appears to result from putting certain code into proper ANF. The most concise example so far comes from an example of dynamic wind

```racket
(define val
    (let* ([x (make-parameter 0)]
           [l null]
           [add (lambda (a b) (set! l (append l (list (cons a b)))))])
        (let ([k (parameterize ([x 5])
                   (dynamic-wind
                       (lambda () (display "pre: ") (displayln (x)) (add 1 (x)))
                       (lambda () (parameterize ([x 6])
                                    (display "mid: ") (displayln (x))
                                    (let ([k+e (let/cc k (cons k void))])
                                      (add 2 (x))
                                      ((cdr k+e))
                                      (car k+e))))
                       (lambda () (display "post ") (displayln (x)) (add 3 (x)))))])
          (parameterize ([x 7])
            (let/cc esc
              (k (cons void esc))))
          )
        l)
    )

val ;; => ((1 . 5) (2 . 6) (3 . 5) (1 . 5) (2 . 6) (3 . 5))
(equal? val '((1 . 5) (2 . 6) (3 . 5) (1 . 5) (2 . 6) (3 . 5))) ;; => #t
```

An ANF pass will (among other things) put the final reference to `l` into the body of second `let` expression like so

```racket
(define val
    (let* ([x (make-parameter 0)]
           [l null]
           [add (lambda (a b) (set! l (append l (list (cons a b)))))])
        (let ([k (parameterize ([x 5])
                   (dynamic-wind
                       (lambda () (display "pre: ") (displayln (x)) (add 1 (x)))
                       (lambda () (parameterize ([x 6])
                                    (display "mid: ") (displayln (x))
                                    (let ([k+e (let/cc k (cons k void))])
                                      (add 2 (x))
                                      ((cdr k+e))
                                      (car k+e))))
                       (lambda () (display "post ") (displayln (x)) (add 3 (x)))))])
          (parameterize ([x 7])
            (let/cc esc
              (k (cons void esc))))
          l)
        )
    )

;; For Pycket
val ;; => ((1 . 5) (2 . 6) (3 . 5) (1 . 5) (2 . 6) (3 . 7))
(equal? val '((1 . 5) (2 . 6) (3 . 5) (1 . 5) (2 . 6) (3 . 5))) ;; => #f
```

This change does not alter the behaviour of Racket at all, but results in the wrong answer in Pycket. The invocation of the caputred continuation puts the wrong value at the end of the list (`(3 . 7)` instead of `(3 . 5)`).

**Comment by sabauma** [comment]
This probably means we are executing the post action of `dynamic-wind` in the wrong continuation frame.

**Comment by sabauma** [comment]
Fixed by: 246c9f7a31ed40c0f1fc69bba221a3e6c6dc89d0
